File: plugins/sample-text-editor/text_editor.py
"""
Sample text editor plugin for PyThra Framework

This is a demonstration of the new package system with enhanced metadata
and discovery capabilities.
"""

class TextEditorPlugin:
    """Simple text editor plugin"""
    
    def __init__(self):
        self.name = "sample-text-editor"
        self.version = "1.0.0"
    
    def get_widget_definition(self):
        """Return the widget definition for the text editor"""
        return {
            "name": "TextEditor",
            "component": "TextEditorWidget",
            "props": {
                "placeholder": "Enter text here...",
                "multiline": True,
                "readonly": False
            }
        }
    
    def initialize(self, framework):
        """Initialize the plugin with the PyThra framework"""
        print(f"Initializing {self.name} v{self.version}")
        return True

# Plugin entry point
def create_plugin():
    """Factory function to create plugin instance"""
    return TextEditorPlugin()

File: plugins/sample-text-editor/text_editor.js
/**
 * TextEditor Widget - PyThra JavaScript Component
 * 
 * This is a sample JavaScript widget component that demonstrates
 * the new package system integration.
 */

class TextEditorWidget {
    constructor(element, props = {}) {
        this.element = element;
        this.props = {
            placeholder: 'Enter text here...',
            multiline: true,
            readonly: false,
            ...props
        };
        
        this.init();
    }
    
    init() {
        // Create the textarea element
        this.textarea = document.createElement(this.props.multiline ? 'textarea' : 'input');
        
        if (!this.props.multiline) {
            this.textarea.type = 'text';
        }
        
        this.textarea.placeholder = this.props.placeholder;
        this.textarea.readOnly = this.props.readonly;
        this.textarea.className = 'pythra-text-editor';
        
        // Add some basic styling
        this.textarea.style.cssText = `
            width: 100%;
            min-height: 200px;
            padding: 12px;
            border: 1px solid #ddd;
            border-radius: 4px;
            font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
            font-size: 14px;
            resize: vertical;
            outline: none;
        `;
        
        // Add focus styling
        this.textarea.addEventListener('focus', () => {
            this.textarea.style.borderColor = '#0066cc';
            this.textarea.style.boxShadow = '0 0 0 2px rgba(0, 102, 204, 0.2)';
        });
        
        this.textarea.addEventListener('blur', () => {
            this.textarea.style.borderColor = '#ddd';
            this.textarea.style.boxShadow = 'none';
        });
        
        // Add to DOM
        this.element.appendChild(this.textarea);
    }
    
    getValue() {
        return this.textarea.value;
    }
    
    setValue(value) {
        this.textarea.value = value;
    }
    
    focus() {
        this.textarea.focus();
    }
    
    destroy() {
        if (this.textarea) {
            this.textarea.remove();
        }
    }
}

// Export for PyThra framework
window.TextEditorWidget = TextEditorWidget;